index.js

```diff
@@ -439,52 +453,29 @@ Router.prototype.process_params = function process_params(layer, called, req, re
  * @public
  */
 
-Router.prototype.use = function use(handler) {
-  var offset = 0
-  var path = '/'
-
-  // default path to '/'
-  // disambiguate router.use([handler])
-  if (typeof handler !== 'function') {
-    var arg = handler
-
-    while (Array.isArray(arg) && arg.length !== 0) {
-      arg = arg[0]
-    }
-
-    // first arg is the path
-    if (typeof arg !== 'function') {
-      offset = 1
-      path = handler
-    }
-  }
-
-  var callbacks = flatten(slice.call(arguments, offset))
-
-  if (callbacks.length === 0) {
-    throw new TypeError('argument handler is required')
-  }
-
-  for (var i = 0; i < callbacks.length; i++) {
-    var fn = callbacks[i]
+Router.prototype.use = function use() {
+  var args = processUseArgs(arguments)
+  this.stack = this.stack.concat(createLayers(args[0], args[1], {
```

Comment:
This `.concat` seems to be undocumented in the `mcstack` repo.

Reply:
Ok, yeah so that is just because it extends array.  Would it satisfy if I added a section at the top of the api docs which linked to the array documentation and said "Stack and FlatStack extend from array"?

Reply:
I get that it extends an array, but there is no documentation for what that array actually contains. Is it an array of strings, numbers, objects of some kind, functions... ? I cannot tell if the result of `createLayers` where results in something that should be in that array.

Reply:
It is an array of functions, but I guess I can see how this line abstracts that in such a way where it is hard to know.  Let me see if there is a way I can make this more clear in the documentation.  `createLayers` returns the result of `callbacks.map`, so it returns an array of middleware functions which are just passed to `Array.prototype.concat`.

Reply:
So if `.concat` should be given an array of function, then this API is used wrong here. The `createLayers` function is returning an array of `Layer` objects, not an array of functions.

Reply:
I can make `router.stack` a plain array again, it just seemed reasonable to use a stack here since it works just fine.

Reply:
Yeah, Now that I read this again I think it is better to keep them separate.  So I changed `router.stack` back to a plain array.  I will push once we have any other changes worked out (so I don't break any reviews in progress).

Reply:
I haven't dug into the PR deeply to comment on what makes sense or not; currently my comments are just about if we're splitting things apart into different modules, then we're careful to use the external module as is documented/support by said module so we don't end up breaking ourselves on accident with mysterious API usage is all.

Reply:
Perfectly understandable.  I will make sure that `mwstack` has a more clear documentation on the fact that it extends `Array`.  In this case, now that I am removing it's usage as `router.stack` we are only importing it to expose it and use the one documented method `mwStack.isError`.  Seems like that solves the concern?

Reply:
Perhaps. I'm not sure without seeing it to honestly say for sure 😆 . To be clear: my concern is just that we're using a module as that module documents being used. The `isError` seems to say the following from the `mwstack` docs:

> Returns true when the middleware was added to a stack as an error middleware.

It seems to imply, at least to me, that it is to be used on a `func` that was added to the `Stack` of this module, is that right? And it will tell you yes/no if it was added to `Stack` as an error middleware or not?

Reply:
If it is used on a function which was never added to a stack as an error it will always return `false` (so just any random mw function).  But yes, it is intended to be passed any middleware function and will return `true`/`false` accordingly.